### PR TITLE
Add group device_class

### PR DESCRIPTION
**Pull request by twhittock**
## Proposed change
group_device_class PR adds device_class functionality.

Example and configuration documentation added here.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/36324
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by twhittock** [comment]
I rebased it on the 'next' branch. Unassigning myself.

File: source/_integrations/group.markdown [hunk 1]
```diff
@@ -51,6 +52,10 @@ icon:
   description: The icon that shows in the front end.
   required: false
   type: string
+device_class:
+  description: The device class which the group matches.
```
Comment (klaasnicolaas) [inline source/_integrations/group.markdown]: Maybe make it clear to the user which device classes you can choose? Link to a page with the list of options?

Comment (twhittock) [reply source/_integrations/group.markdown]: Sorry, I don't know the syntax to link to the binary sensor device list page. Can you help point me in the right direction?

Comment (klaasnicolaas) [reply source/_integrations/group.markdown]:
```suggestion
  description: The device class which the group matches, list of possible [device classes](/integrations/binary_sensor/#device-class).
```

**Pull request closed without merging**